website/src/lib/oasst_inference_client.ts

```diff
@@ -0,0 +1,61 @@
+import axios, { AxiosRequestConfig } from "axios";
+import Cookies from "cookies";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { JWT } from "next-auth/jwt";
+import { InferenceCreateChatResponse, InferenceDebugTokenResponse } from "src/types/Chat";
+
+// TODO: this class could be structured better
+export class OasstInferenceClient {
+  private readonly cookies: Cookies;
+  private inferenceToken: string;
+  private readonly userTokenSub: string;
+
+  constructor(req: NextApiRequest, res: NextApiResponse, token: JWT) {
```

Comment:
Depending on req and res in this class is really ugly IMO. We can extract to a custom factory or something. But I'm fine with it for now as it's just a prototype.

Reply:
Totally agree, I did not find a cleaner solution:

I wanted to save the user's auth token in the session of the user, I tried to inject into the JWT but it seems that next-auth only allows this in very restrictive location (i.e. only when creating a session).

The other option was cookies, the reason we depend on `req` and `res` is to read and write the cookie, which is handled by the `cookies` library, and it does not expose a cleaner API other than giving the req / res directly in the constructor.

We can definitely pass the `Cookies` instance as a constructor parameter, this will only partially solve the problem.

Another options we might consider: saving the token in memory since it is temporary anyway, or saving it in the database with the remaining user-related stuff (which will incur a db call overhead per-request)